cypress/support/pages/course/pageShareCourseModal.js

```diff
@@ -0,0 +1,63 @@
+"use strict"
+
+class ShareCourseModal {
+	static #shareCourseDialog = '[data-testid="share-dialog"]'
+	static #shareCourseDialogInfoTextTitle = '[data-testid="share-options-info-text"]';
+	static #shareCourseDialogInfoTextContainer = '[data-testid="share-options-table-header"]';
+	static #shareCourseDialogPersonalDataInfo = '[data-testid="share-options-personal-data-text"]';
+	static #shareCourseDialogExternalToolsInfo = '[data-testid="share-modal-external-tools-info"]';
+	static #shareCourseDialogSchoolInternalCheckBox = '[data-testid="isSchoolInternal"]';
+	static #shareCourseDialogExpiryDateCheckBox = '[data-testid="hasExpiryDate"]';
+	static #shareCourseResultUrlTextBox = '[data-testid="share-course-result-url"]';
+	static #dialogNextButton = '[data-testid="dialog-next"]';
+
+	seeShareCourseDialogBox() {
+		cy.get(ShareCourseModal.#shareCourseDialog).should("exist");
+	}
+
+	seeInfoTextInShareCourseDialog() {
+		cy.get(ShareCourseModal.#shareCourseDialogInfoTextTitle).should("exist");
```

Comment:
same here for alls tests with exists. use should("be.visible")

add test for protected parameter info text
- data-testid="share-modal-external-tools-info"
the data-testid is already implemented in the vue-client